cooltools/cli/compute_expected.py

```diff
@@ -91,52 +85,60 @@ def compute_expected(cool_path, nproc, chunksize, output, hdf, contact_type, wei
     """
     clr = cooler.Cooler(cool_path)
     supports = [(chrom, 0, clr.chromsizes[chrom]) for chrom in clr.chromnames]
-    weight1 = weight_name+"1"
-    weight2 = weight_name+"2"
 
+    # define transofrms - balanced and raw ('count') for now
+    if balanced:
+        weight1 = weight_name+"1"
+        weight2 = weight_name+"2"
+        transforms = {
+            'balanced': lambda p: p['count'] * p[weight1] * p[weight2] }
+    else:
+        transforms = {}
+
+    # execution details
     if nproc > 1:
         pool = mp.Pool(nproc)
         map_ = pool.map
     else:
         map_ = map
 
+    # using try-clause to close mp.Pool properly
     try:
         if contact_type == 'cis':
             tables = expected.diagsum(
                 clr,
                 supports,
-                transforms={
-                    'balanced': lambda p: p['count'] * p[weight1] * p[weight2]
-                },
+                transforms=transforms,
                 chunksize=chunksize,
-                ignore_diags=drop_diags,
+                ignore_diags=ignore_diags,
                 map=map_)
             result = pd.concat(
                 [tables[support] for support in supports],
                 keys=[support[0] for support in supports],
                 names=['chrom'])
-            result['balanced.avg'] = result['balanced.sum'] / result['n_valid']
             result = result.reset_index()
 
         elif contact_type == 'trans':
             records = expected.blocksum_pairwise(
                 clr,
                 supports,
-                transforms={
-                    'balanced': lambda p: p['count'] * p[weight1] * p[weight2]
-                },
+                transforms=transforms,
                 chunksize=chunksize,
                 map=map_)
             result = pd.DataFrame(
                 [{'chrom1': s1[0], 'chrom2': s2[0], **rec}
                     for (s1, s2), rec in records.items()],
                 columns=['chrom1', 'chrom2', 'n_valid',
                          'count.sum', 'balanced.sum'])
-            result['balanced.avg'] = result['balanced.sum'] / result['n_valid']
     finally:
         if nproc > 1:
             pool.close()
 
+    # calculate actual averages by dividing sum by n_valid:
+    result['count.avg'] = result['count.sum'] / result['n_valid']
```

Comment:
related - just realized that this https://github.com/mirnylab/cooltools/blob/cb77cf219aa38286f418b0a15eb44128de295d4e/cooltools/expected.py#L775
prevents us from calculating `trans` expected for unbalanced data ...

I know using `n_total` alone overestimates denominator, but maybe we could allow that in case someone really needs that ?

application in mind: very low resolution data ~1-5Mb bins for shallow data, e.g. for looking for translocations and other Structural Variation in cancer cell lines etc - in this cases balancing might not be applicable to begin with ...
what do you think ?